File: examples/proxy/rest/csharp/ProxyExample.cs
// Routes browser traffic through a residential proxy and prints the resulting IP.
//
// Run: dotnet run

using System.Net.Http;
using System.Text;
using System.Text.Json;

string token = "YOUR_API_TOKEN_HERE";
string endpoint = $"https://production-sfo.browserless.io/scrape?token={token}&proxy=residential&proxyCountry=us";

var payload = new
{
    url = "https://api.ipify.org?format=json",
    elements = new[] { new { selector = "body" } },
};

using (HttpClient httpClient = new HttpClient())
{
    var content = new StringContent(
        JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");
    var response = await httpClient.PostAsync(endpoint, content);
    string body = await response.Content.ReadAsStringAsync();
    Console.WriteLine(body);
}

File: examples/proxy/rest/java/ProxyExample.java
// Routes browser traffic through a residential proxy and prints the resulting IP.
//
// Run: javac ProxyExample.java && java ProxyExample

import java.net.URI;
import java.net.http.*;

public class ProxyExample {
    public static void main(String[] args) throws Exception {
        String token = "YOUR_API_TOKEN_HERE";
        String endpoint = "https://production-sfo.browserless.io/scrape?token=" + token
            + "&proxy=residential&proxyCountry=us";

        String payload = "{\"url\": \"https://api.ipify.org?format=json\", \"elements\": [{\"selector\": \"body\"}]}";

        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
            .uri(URI.create(endpoint))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(payload))
            .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response.body());
    }
}
